from_mwt_ds/DataScience/estimate.py

```diff
@@ -0,0 +1,15 @@
+import pandas as pd
+from estimators import bandits
+from estimators.bandits import ips
+
+def estimate_bucket(bucket: pd.DataFrame, config: dict) -> dict:
```

Comment:
Let me put requirements here as separate comment.
1) bucket - data frame that contains the following columns:
    a. 'p' - probability of decision that were made by the online policy
    b. 'r' - reward that was observed online
    c. number of other columns that represent policies that we would love to estimate counterfactually
2) config - configuration object that is dictionary that maps policy names (from 1/c.) to list of estimator names (the ones defined in estimator library)
3) for every policy_name-estimator pair from the config the function should:
- generate the estimate_name (i.e. by concatenating of policy_name and estimator name)
- create an instance of certain estimator
- for all rows in bucket dataframe feed (row['p'], row['r'], row[policy_name]) triplets into the estimator
- add (estimate_name: estimator) tuple to final result

Reply:
Looks like we have most pieces of it, but there are still several gaps:
- although we have proper loop of iterating over policy_name-estimator pairs, we are iterating over rows separately while we can notice that it should triple nested loop.
- we always create ips estimator instead of creating estimators specified in the config
- right now we are just iterating over data without feeding them into estimator objects, so we are not producing results.

Reply:
I think what I'm struggling with specifically is how to get the add_example to work in this context. The way it is written ^^ isn't working for me.

Reply:
please find example of iterating over dataframe rows in answers to this topic:
https://stackoverflow.com/questions/16476924/how-to-iterate-over-rows-in-a-dataframe-in-pandas

Reply:
Oh, I think this is the problem. There's an extra index: My output is:
```
(0, p           0.5
r           1.0
policy_1    0.5
Name: 0, dtype: float64)
['ips']
None
```
That first 0 before the p is giving me that key error, I think.

Reply:
just made some changes. Feedback and more detail for the stub welcome. Is that stub supposed to be in that file btw? (not sure). Thanks!